pkg/cloud/services/network/vpc.go

```diff
@@ -202,6 +202,12 @@ func (s *Service) createVPC() (*infrav1.VPCSpec, error) {
 func (s *Service) deleteVPC() error {
 	vpc := s.scope.VPC()
 
+	// Skip deletion if .spec.vpc.id is not set.
+	if vpc.ID == "" {
```

Comment:
Below we return if VPC does not exist:

> 		if code, ok := awserrors.Code(err); ok && code == awserrors.VPCNotFound {

When the ID is nil, don't we get VPCNotFound error?

Reply:
Since we are getting below error in this scenario:

> failed to delete vpc "": UnauthorizedOperation: You are not authorized to perform this operation.

which is arising from [here](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/pkg/cloud/scope/clients.go#L75), it doesn't throw `VPCNotFound` error.

Reply:
In this issue, it is stated that this error is seen due to auth issues.

Are you able to replicate this locally and saw the same error?

Reply:
I have not replicated the issue locally yet, will do and provide the findings.

Reply:
makes sense to me, I will close the PR and issue for now and maybe we can reopen if we face this issue again.